wordpress-mobile/WordPress-Android
# Refactor StateListItemBuilders

Parent #13328 

This PR is the last part of a consolidation effort of the Jetpack backup/restore features. This part focuses streamlining the state list builders. 

I replaced the multiple classes with a single `RestoreStateListItemBuilder` for Restore functionality and `BackupDownloadStateListItemBuilder` for backup download functionality. I intentionally kept the builders separate by feature so that commonalities within the feature itself can be easily utilized.

Recap
- Two new classes
	- RestoreStateListItemBuilder
	- BackupDownloadStateListItemBuilder
- Test classes include import changes only
- ViewModel classes include import changes only
- Deleted 7 classes

For reference: Part I #13788, Part II #13789, Part III #13790


**To test:**
- Run the app and make sure the backup/download/restore still function as expected

PR submission checklist:
